File: mobile-ui-testing-android/app/src/androidTest/java/biz/tearoom6/mobile_ui_testing_android/uitest/UserLoginTests.java
package biz.tearoom6.mobile_ui_testing_android.uitest;

import android.support.test.runner.AndroidJUnit4;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;

import biz.tearoom6.mobile_ui_testing_android.R;
import biz.tearoom6.mobile_ui_testing_android.uitest.helper.MainActivityTestBase;

import static android.support.test.espresso.matcher.ViewMatchers.assertThat;
import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
import static android.support.test.espresso.matcher.ViewMatchers.withText;
import static biz.tearoom6.mobile_ui_testing_android.uitest.helper.AssetHelper.loadTestUser;
import static biz.tearoom6.mobile_ui_testing_android.uitest.helper.UserHelper.login;
import static biz.tearoom6.mobile_ui_testing_android.uitest.helper.UserHelper.loginAsRegisteredUser;
import static biz.tearoom6.mobile_ui_testing_android.uitest.helper.UserHelper.logout;
import static biz.tearoom6.mobile_ui_testing_android.uitest.helper.ViewHelper.waitUntilTextDisplayed;

@RunWith(AndroidJUnit4.class)
public class UserLoginTests extends MainActivityTestBase {

    @Before
    public void setup() {
        logout();
    }

    @Test
    public void loginRegisteredAccount() {
        loginAsRegisteredUser();

        // Check registered nickname & email is displayed.
        assertThat(
                "Registered nickname is displayed",
                activity.findViewById(R.id.auth_label_nickname),
                withText(loadTestUser().getNickname())
        );
        assertThat(
                "Registered email is displayed",
                activity.findViewById(R.id.auth_label_email),
                withText(loadTestUser().getEmail())
        );

        // Check logout button is displayed.
        assertThat(
                "Logout button is displayed",
                activity.findViewById(R.id.auth_button_logout),
                isDisplayed()
        );
    }

    @Test
    public void loginWithInvalidPassword() {
        // Login as registered user with invalid password.
        login(loadTestUser().getEmail(), "invalid_password");

        waitUntilTextDisplayed(R.id.auth_label_loginErrors);

        // Check error message is displayed properly.
        String expectMessage = messageHelper.getLocalizedMessage(R.string.failed_login);
        assertThat(
                "Error message is displayed properly.",
                activity.findViewById(R.id.auth_label_loginErrors),
                withText(expectMessage)
        );
    }
}

File: mobile-ui-testing-ios/mobile-ui-testing-iosUITests/UserLoginTests.swift
import XCTest

class UserLoginTests: XCTestCase {
    
    let app = XCUIApplication()
    
    override func setUp() {
        super.setUp()
        continueAfterFailure = false
        app.launch()
        
        logout()
    }
    
    override func tearDown() {
        super.tearDown()
    }
    
    func testLoginAsRegisteredAccount() {
        loginAsRegisteredUser()
        
        // Check registered nickname & email is displayed.
        XCTAssertEqual(UserStatus.nicknameLabel.label, RegisteredUserNickname, "Registered nickname is displayed")
        XCTAssertEqual(UserStatus.emailLabel.label, RegisteredUserEmail, "Registered email is displayed")
        
        // Check logout button is displayed.
        XCTAssertTrue(UserStatus.logoutButton.isHittable, "Logout button is displayed")
    }
    
    func testLoginWithInvalidPassword() {
        // Login as registered user with invalid password.
        login(email: RegisteredUserEmail, password: "invalid_password")
        
        waitHittable(for: Login.errorMessageLabel)
        
        // Check error message is displayed properly.
        let expectMessage = Localized().string(forKey: "error_login_wrongPassword")
        XCTAssertEqual(Login.errorMessageLabel.label, expectMessage, "Error message is displayed properly.")
    }
    
}
